plugin/trino-openlineage/src/test/java/io/trino/plugin/openlineage/OpenLineageListenerQueryRunner.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage;
+
+import io.trino.Session;
+import io.trino.plugin.memory.MemoryPlugin;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+
+import java.util.Map;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public final class OpenLineageListenerQueryRunner
+{
+    public static final String CATALOG = "marquez";
+    public static final String SCHEMA = "default";
+
+    private OpenLineageListenerQueryRunner() {}
+
+    public static QueryRunner createOpenLineageRunner(Map<String, String> listenerProperties)
+            throws Exception
+    {
+        QueryRunner queryRunner = null;
```

Comment:
There's no need to initialize separately. 
```java
        QueryRunner queryRunner = DistributedQueryRunner.builder(createSession())
                .setEventListener(new OpenLineageListenerFactory().create(listenerProperties))
                .build();
```

Reply:
but then I won't be able to pass it to closeAllSuppress upon some error no?